frontend/src/wizards/Governance/Policy/PolicyWizard.tsx

```diff
@@ -620,6 +547,183 @@ export function isExistingTemplateName(name: string, policies: IResource[]) {
   return false
 }
 
+function OperatorPolicy() {
+  const { t } = useTranslation()
+  const template: any = useItem()
+
+  const { update } = useData()
+
+  const [allNamespacesMode, setAllNamespacesMode] = useState<boolean>(
+    !template?.objectDefinition?.spec?.operatorGroup?.targetNamespaces
+  )
+
+  function setTargetNamespace(targetNamespace: string) {
+    if (allNamespacesMode) {
+      return
+    }
+
+    if (targetNamespace === '') {
+      removeTargetNamespace()
+
+      return
+    }
+
+    if (!template?.objectDefinition) {
+      return
+    }
+
+    if (!template.objectDefinition?.spec) {
+      template.objectDefinition.spec = { operatorGroup: {} }
+    }
+
+    if (!template.objectDefinition.spec?.operatorGroup) {
+      template.objectDefinition.spec.operatorGroup = {}
+    }
+
+    if (
+      template.objectDefinition.spec.operatorGroup.targetNamespaces &&
+      template.objectDefinition.spec.operatorGroup.targetNamespaces[0] === targetNamespace
+    ) {
+      return
+    }
+
+    template.objectDefinition.spec.operatorGroup.targetNamespaces = [targetNamespace]
+
+    update()
+  }
+
+  function removeTargetNamespace() {
+    if (template?.objectDefinition?.spec?.operatorGroup?.targetNamespaces) {
+      // If the operator group only has targetNamespaces set, then remove the entire operatorGroup section from
+      // the policy.
+      if (Object.keys(template?.objectDefinition?.spec?.operatorGroup).length === 1) {
+        set(template, 'objectDefinition.spec.operatorGroup', undefined)
```

Comment:
what is diff using between set function  and template + update. same result?

Reply:
Looking at the source for `set`, it seems it'd be faster to just set it directly.

Reply:
good to know Thanks!